sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client.py

```diff
@@ -360,6 +360,33 @@ def delete_share(
         except StorageErrorException as error:
             process_storage_error(error)
 
+    @distributed_trace
+    def undelete_share(self, deleted_share_name, deleted_share_version, **kwargs):
+        # type: (str, str, **Any) -> None
+        """Restores soft-deleted share.
+
+        Operation will only be successful if used within the specified number of days
+        set in the delete retention policy.
+
+        :param str deleted_share_name:
+            Specifies the name of the deleted share to restore.
+            Service Version 2019-12-12 and later.
+        :param str deleted_share_version:
+            Specifies the version of the deleted share to restore.
+            Service Version 2019-12-12 and later.
+        :keyword int timeout:
+            The timeout parameter is expressed in seconds.
+        :rtype: None
+        """
+        if deleted_share_name != self.share_name:
```

Comment:
why not directly pass self.share_name?

Reply:
good question.
deleted_share_name is the name when it is deleted, the name on ShareClient is the one you want to restore to. Currently there's a bug in service so that deleted_share_name should be the same as ShareClient.share_name, while in stg74 they could be different